packages/ui/src/components/Modal/components/Dialog.tsx

```diff
@@ -312,8 +335,18 @@ export const Dialog = ({
         top={Math.max(top, 0)}
         data-animation={animation}
         size={size}
+        data-has-image={!!image}
       >
-        {children}
+        {image ? (
+          <>
+            <StyledDiv>
+              <StyledImage src={image} alt="promotional image" />
+            </StyledDiv>
+            <StyledStack gap={5}>{children}</StyledStack>
+          </>
```

Comment:
I know you got the code from the existing one maybe it needs reflexion onto it. I think it might be easier to set the image in CSS with background. 

Do we want to have the image to take the full space or to be centered like you did ?
Might be worth questioning the design about this and adapt the code depending on that. 
<img width="796" alt="Screenshot 2025-04-03 at 10 21 09" src="https://github.com/user-attachments/assets/17594850-c7f7-4c39-a4d8-af4287453dd3" />

The original component was done for a specific use case, here we will try to make it more generic.